contentgrid-spring-data-rest/src/main/java/org/springframework/data/rest/webmvc/DelegatingRepositoryPropertyReferenceController.java

```diff
@@ -127,14 +131,27 @@ public ResponseEntity<?> followPropertyReference(final RootResourceInformation r
         );
     }
 
-    private static Optional<String> findReverseRelationPropertyName(PersistentProperty<?> property) {
+    private Optional<String> findReverseRelationPropertyName(PersistentProperty<?> property) {
         var oneToMany = property.findAnnotation(OneToMany.class);
         if (oneToMany != null) {
             // when this side is the inverse side of a bi-directional relation
             if (!"".equals(oneToMany.mappedBy())) {
                 return Optional.of(oneToMany.mappedBy());
             }
 
+            // It's also possible there is an implicit relation with the other side being a plain @Column that matches our @JoinColumn
+            var joinColumn = property.findAnnotation(JoinColumn.class);
+            if(joinColumn != null) {
+                var ourColumnName = joinColumn.name();
+                var inverseSide = repositories.getPersistentEntity(property.getAssociationTargetType());
+                for (PersistentProperty<? extends PersistentProperty<?>> inverseProperty : inverseSide.getPersistentProperties(
+                        Column.class)) {
+                    if (Objects.equals(ourColumnName, inverseProperty.findAnnotation(Column.class).name())) {
```

Comment:
`findAnnotation` returns an Optional, you need to destructure it before the equality check.

Reply:
1. `Property#findAnnotation()` returns the value or null.
2. Given we use `getPersistentProperties(Column.class)`, we only get properties that have a `@Column` annotation, so `#findAnnotation()` will never return null.

Reply:
Yeah it became clear in intellij. For some reason, github's popup code thingy when clicking the method assured me that it returns an Optional.